fast_carpenter/summary/binned_dataframe.py

```diff
@@ -7,17 +7,30 @@
 
 
 class Collector():
-    def __init__(self, filename, dataset_col, binnings):
+    def __init__(self, filename, dataset_col, binnings, file_format):
         self.filename = filename
         self.dataset_col = dataset_col
         self.binnings = binnings
+        self.file_format = file_format
 
     def collect(self, dataset_readers_list):
         if len(dataset_readers_list) == 0:
             return
 
         output = self._prepare_output(dataset_readers_list)
-        output.to_csv(self.filename, float_format="%.17g")
+
+        valid_ext = {'xlsx': 'excel', 'h5': 'hdf', 'msg': 'msgpack', 'dta': 'stata', 'pkl': 'pickle', 'p': 'pickle'}
```

Comment:
I think this could become a class attribute rather than something defined within the function.  The main reason is it allows monkeypatching by a user if they want to extend this (which can make testing easier, for example). So then move this line to just below `class Collector():` and then reference `valid_ext` as `Collector.valid_ext` in the code.

Reply:
Yeah that's a good idea, i'll move it to a class attribute